enhancements/sig-architecture/18-workload-scheduling/README.md

```diff
@@ -0,0 +1,220 @@
+
+### Non Goals
+
+- This KEP will not cover the distribution of special workloads such as `Job`.
+- This KEP will not cover the distribution of satellite resources around the workload 
+  such as `ConfigMap`, `ServiceAccount`.
+
+## Proposal
+
+### Abstraction
+
+To understand the functionalities of the multi-cluster workload easier, we can start by
+defining the boundary of the controller's abstraction as a black box.
+
+#### "ClusterSet" and "Placement"
+
+#### Controller "Input"
+
+The source of information input for the multi-cluster workload controller will be:
+
+- __Cluster Topology__: The `PlacementDesicision` is dynamically computed according to
+  the following knowledge from OCM's built-in APIs:
+  - The existence and availability of the managed clusters.
+  - The "hub-defined" attributes attached to the cluster model via labelling. e.g. the 
+    clusterset, the feature label or other custom labels which will be read by the 
+    placement controller.
+  - The "spoke-reported" attributes i.e. "ClusterClaim" which is collected and reported
+    by the spoke agent.
+    
+- __API Prescription__: There will be a new API named "ElasticWorkload" or 
+  "ManagedWorkload" that prescribes the necessary information for workload distribution
+  such as the content of the target workload, the expected total number of replicas, etc.
+  
+- __Effective Distributed Local Workload__: The new controller also need to capture the
+  events from local clusters so that it can take actions e.g. when the instance crushes
+  or tainted unexpectedly.
+
+#### Controller "Output"
+
+The new controller will be applying the latest state of the workload towards the selected
+clusters and tuning its replicas on demand. As a matter of implementation, the workload
+applying will be executed via the stable `ManifestWork` api.
+
+### API Spec
+
+```yaml
+apiVerion: scheduling.open-cluster-management.io/v1
+kind: ElasticWorkload
+spec:
+  # The target namespace to deploy the workload in the spoke cluster.
+  spokeNamespace: default
+  # The content of target workload, supporting:
+  # - Inline: Embedding a static manifest.
+  # - Import: Referencing an existing workload resource. (Note that
+  #           the replicas should always be set to 0 to avoid wasting
+  #           capacities in the hub cluster.)
+  target:
+    type: [ Inline | Import ]
+    inline: ...
+    import: ...
+  # Referencing an OCM's placement policy in the same namespace as where
+  # this elastic workload resource lives.
+  placementRef:
+    name: ...
+  # DistributionStrategy controls the expected replicas distribution
+  # across the selected clusters from the placement api above. The supported 
+  # distributing strategy will be:
+  # - Even: Filling the min replicas upon every round. i.e. max-min
+  # - Weighted: Setting a default weight and overriding the weight for a
+  #             few clusters on demand.
+  distributionStrategy:
+    totalReplicas: 10
+    type: [ Even | Weighted ]
```

Comment:
how to specify cluster weight, at "Placement.prioritizerPolicy"?

Reply:
for clarification "Placement.prioritizerPolicy" only takes effect during cluster selection, while the `Weighted` distribution indicates the distribution of replicas for the workload. the a sample of weighted distribution will be something like:

```
spec:
  distributionStrategy:
    totalReplicas: 10
    type: Weighted
    weighted:
      defaultWeight: 10
      overrides:
      - clusterName: xx
         weight: 100
```